server/repo.go

```diff
@@ -235,3 +220,53 @@ func RepairRepo(c *gin.Context) {
 	}
 	c.Writer.WriteHeader(http.StatusOK)
 }
+
+func MoveRepo(c *gin.Context) {
+	remote := remote.FromContext(c)
+	repo := session.Repo(c)
+	user := session.User(c)
+
+	to, exists := c.GetQuery("to")
+	if !exists {
+		err := fmt.Errorf("Missing required to query value")
+		c.AbortWithError(http.StatusInternalServerError, err)
+	}
+
+	owner, name, errParse := ParseRepo(to)
+	if errParse != nil {
+		c.AbortWithError(http.StatusInternalServerError, errParse)
+	}
+
+	from, err := remote.Repo(user, owner, name)
+	if err != nil {
+		c.AbortWithError(http.StatusInternalServerError, err)
+	}
+	repo.Name = from.Name
+	repo.Owner = from.Owner
+	repo.FullName = from.FullName
+	repo.Avatar = from.Avatar
+	repo.Link = from.Link
+	repo.Clone = from.Clone
+	repo.IsPrivate = from.IsPrivate
+	repo.Visibility = from.Visibility
```

Comment:
Sounds good I'll add since I'll need to use it when I create a PR to have patch behave the same way